src/chemistry/geoschem/cesmgc_history_mod.F90

```diff
@@ -0,0 +1,1232 @@
+                                RC        = RC                      )
+       ELSEIF ( current%isChem ) THEN
+          CALL Registry_Lookup( am_I_Root = am_I_Root,               &
+                                Registry  = State_Chm%Registry,      &
+                                RegDict   = State_Chm%RegDict,       &
+                                State     = State_Chm%State,         &
+                                Variable  = current%registryID,      &
+                                Ptr2d_4   = current%GCStateData2d_4, &
+                                Ptr2d_8   = current%GCStateData2d_8, &
+                                Ptr2d_I   = current%GCStateData2d_I, &
+                                Ptr3d_4   = current%GCStateData3d_4, &
+                                Ptr3d_8   = current%GCStateData3d_8, &
+                                Ptr3d_I   = current%GCStateData3d_I, &
+                                RC        = RC                      )
+       ELSEIF ( current%isDiag ) THEN
+          CALL Registry_Lookup( am_I_Root = am_I_Root,               &
+                                Registry  = State_Diag%Registry,     &
+                                RegDict   = State_Diag%RegDict,      &
+                                State     = State_Diag%State,        &
+                                Variable  = current%registryID,      &
+                                Ptr2d_4   = current%GCStateData2d_4, &
+                                Ptr2d_8   = current%GCStateData2d_8, &
+                                Ptr2d_I   = current%GCStateData2d_I, &
+                                Ptr3d_4   = current%GCStateData3d_4, &
+                                Ptr3d_8   = current%GCStateData3d_8, &
+                                Ptr3d_I   = current%GCStateData3d_I, &
+                                RC        = RC                      )
+       ENDIF
+       IF ( RC == GC_FAILURE ) THEN
+          ErrMsg = "Registry pointer not found for " // TRIM(current%name) // &
+                   ". Check that the tag (e.g. species) is valid "         // &
+                   "for this diagnostic."
+          EXIT
+       ENDIF
+
+       ! For CESM export, outfld accepts the data pointer directly but it
+       ! has to be in r8. Thus, allocate a r8 data type in exportData2d or
+       ! ExportData3d so that the rest of the code can be reused, then
+       ! update pointer data can just call outfld additionally. There is,
+       ! however, a memory hit from this. Revisit later. (hplin, 10/31/22)
+       !
+       ! As a side note, in CESM-GC, State_Grid%NX is always 1 since the data
+       ! is chunkized. Only the State_Grid%NY matters here.
+       !
+       ! Because NY could vary across chunk sizes, allocate one extra column
+       ! but only read up to the actual :State_Grid%NY. This allows for different
+       ! instances of GEOS-Chem to share the same ExportData allocation in separate
+       ! calls to the pointer update subroutine.
```

Comment:
Sorry for the lack of clarity. The number of columns in each chunk may vary by up to 1, I think, so I allocated one extra to account for the largest size. This is for safety's sake. I don't actually recall the exact behavior, but it may depend on the number of CPU cores and how CAM decides to split the domain up.